src/View/Helper/SearchHelper.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Search\View\Helper;
+
+use Cake\View\Helper;
+use Cake\View\View;
+
+/**
+ * @property \Cake\View\Helper\UrlHelper $Url
+ * @property \Cake\View\Helper\HtmlHelper $Html
+ */
+class SearchHelper extends Helper
+{
+    /**
+     * @var array
+     */
+    public $helpers = [
+        'Url',
+        'Html',
+    ];
+
+    /**
+     * Default config for this class
+     *
+     * - 'additionalBlacklist': Additional params that also should be filtered out.
+     *   For pagination views that usually is also limit and page, as the offset would be wrong.
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'additionalBlacklist' => [],
+    ];
+
+    /**
+     * Checks for pagination and if so, blacklist limit and page params.
+     *
+     * @param \Cake\View\View $View View
+     * @param array $config Config
+     */
+    public function __construct(View $View, array $config)
+    {
+        $this->request = $View->request;
+        if ($this->request->getParam('paging')) {
+            $this->_defaultConfig['additionalBlacklist'] = ['page'];
```

Comment:
`$this->_defaultConfig['additionalBlacklist'][] = 'page'` would be a tad better, just in case someone wants to extend the helper and override `$_defaultConfig`.

Reply:
Interesting edge case, sure thing.